packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java

```diff
@@ -247,15 +248,32 @@ public void requestPermissions(
   }
 
   public void onRequestPermissionsResult(
-      final int requestCode, final String[] permissions, final int[] grantResults) {
-    mPermissionsCallback =
-        args -> {
-          if (mPermissionListener != null
-              && mPermissionListener.onRequestPermissionsResult(
-                  requestCode, permissions, grantResults)) {
-            mPermissionListener = null;
-          }
-        };
+    final int requestCode, final String[] permissions, final int[] grantResults) {
+    Callback permissionsCallback = args -> {
+      if (mPermissionListener != null
+        && mPermissionListener.onRequestPermissionsResult(
+        requestCode, permissions, grantResults)) {
+        mPermissionListener = null;
+      }
+    };
+
+    LifecycleState lifecycle;
+    if (isFabricEnabled()) {
+      ReactHost reactHost = getReactHost();
+      lifecycle = reactHost != null ? reactHost.getLifecycleState() : LifecycleState.BEFORE_CREATE;
+    } else {
+      ReactInstanceManager instanceManager = getReactInstanceManager();
```

Comment:
Shouldn't we access the reactInstanceManager from `reactNativeHost`? 

```suggestion
      ReactInstanceManager instanceManager = getReactNativeHost().getReactInstanceManager();
```

Reply:
Done